Added support for git.centos.org

Rajeshpatel07 opened the pull request:
fixes #153 
 Added a new option `--pfm` ( platform ) for centos and pagure platform.

gridhead commented on pagure_exporter/work/tkts.py:
```diff
@@ -199,12 +211,18 @@ def itercmts(self, dictobjc):
-                yy=datetime.fromtimestamp(standard.cmtstime, timezone.utc).strftime("%Y"),
-                hh=datetime.fromtimestamp(standard.cmtstime, timezone.utc).strftime("%H"),
-                mm=datetime.fromtimestamp(standard.cmtstime, timezone.utc).strftime("%M"),
+                dateinfo=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%c"),
+                mo=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%b").lower(),
+                dd=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%d"),
+                yy=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%Y"),
+                hh=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%H"),
+                mm=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%M"),
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/work/tkts.py:
```diff
@@ -213,9 +231,11 @@ def itercmts(self, dictobjc):
             """
             rqstdata = {"body": bodydata.replace("@", "&")}
-            rslt = standard.gpro.issues.get(id=standard.gtlbtkid).discussions.create(data=rqstdata)
+            rslt = standard.gpro.issues.get(
+                id=standard.gtlbtkid).discussions.create(data=rqstdata)
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/work/tkts.py:
```diff
@@ -213,9 +231,11 @@ def itercmts(self, dictobjc):
+                id=standard.gtlbtkid).discussions.create(data=rqstdata)
             respcode, respresn = (
-                201, f"{standard.destdict['repolink']}/-/issues/{standard.gtlbtkid}#note_{rslt.id}"
+                201, f"{standard.destdict['repolink']
+                        }/-/issues/{standard.gtlbtkid}#note_{rslt.id}"
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/main.py:
```diff
@@ -31,6 +31,14 @@
 
 @click.group(name="pagure_exporter")
+@click.option(
+    "-gp",
```
This needs to be a single character like those mentioned in other options.

gridhead commented on pagure_exporter/main.py:
```diff
@@ -159,7 +167,8 @@ def main(srce, dest, pkey, gkey, fusr, tusr):
 def main_transfer_tkts(status, select, ranges, comments, labels, commit, secret, series):
     if select is not None and ranges is not None:
-        raise click.UsageError("The `select` and `ranges` options cannot be used together")
+        raise click.UsageError(
+            "The `select` and `ranges` options cannot be used together")
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/view/tkts.py:
```diff
@@ -45,7 +45,8 @@ def showtkts():
         if qantrslt[0] == 200:
             general(
-                f"Found {standard.tktcount} issue ticket(s) across {standard.pageqant} "
+                f"Found {standard.tktcount} issue ticket(s) across {
+                    standard.pageqant} "
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/view/tkts.py:
```diff
@@ -55,18 +56,22 @@ def showtkts():
                 if pagerslt[0] == 200:
                     general(
-                        f"Found {len(standard.pagerslt)} issue ticket(s) on this "
+                        f"Found {len(standard.pagerslt)
+                                 } issue ticket(s) on this "
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/view/tkts.py:
```diff
@@ -55,18 +56,22 @@ def showtkts():
                             f"Migrating issue ticket {'with' if standard.movetags else 'without'} "  # noqa: E501
-                            f"labels #{standard.issuiden} '{standard.issuname}' by "
-                            f"'{standard.authname} (ID {standard.authorid})'..."
+                            f"labels #{standard.issuiden} '{
+                                standard.issuname}' by "
+                            f"'{standard.authname} (ID {
+                                                    standard.authorid})'..."
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/view/tkts.py:
```diff
@@ -175,15 +187,17 @@ def showtkts():
                                 if cmtsrslt[0] == 201:
                                     general(
-                                        f"Transferred to {cmtsrslt[1]} in {cmtsrslt[2]} second(s)"
+                                        f"Transferred to {cmtsrslt[1]} in {
+                                            cmtsrslt[2]} second(s)"
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/work/repo.py:
```diff
@@ -97,7 +97,8 @@ def tnfsrepo(self):
                 for brdx in standard.sbrcavbl:
                     repoobjc.git.checkout("%s" % brdx)
-                    repoobjc.git.push(standard.nrmtname, "--set-upstream", brdx, "--force")
+                    repoobjc.git.push(standard.nrmtname,
+                                      "--set-upstream", brdx, "--force")
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/work/tkts.py:
```diff
@@ -133,7 +135,8 @@ def itertkts(self, dictobjc):
             standard.isclosed = True if dictobjc["status"] == "Closed" else False
             standard.authname = dictobjc["user"]["fullname"]
-            standard.authlink = f'{standard.fedora_acc}/{dictobjc["user"]["url_path"]}'
+            standard.authlink = f'{
+                standard.fedora_acc}/{dictobjc["user"]["url_path"]}'
```
Please do not change the codebase that is not associated with the expected change.

gridhead commented on pagure_exporter/work/tkts.py:
```diff
@@ -152,20 +155,27 @@ def itertkts(self, dictobjc):
-                yy=datetime.fromtimestamp(standard.cmtstime, timezone.utc).strftime("%Y"),
-                hh=datetime.fromtimestamp(standard.cmtstime, timezone.utc).strftime("%H"),
-                mm=datetime.fromtimestamp(standard.cmtstime, timezone.utc).strftime("%M"),
+                dateinfo=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%c"),
+                mo=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%b").lower(),
+                dd=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%d"),
+                yy=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%Y"),
+                hh=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%H"),
+                mm=datetime.fromtimestamp(
+                    standard.cmtstime, timezone.utc).strftime("%M"),
```
Please do not change the codebase that is not associated with the expected change.

gridhead reviewed (changes requested):
Please make the suggested changes.

gridhead reviewed (changes requested):
Also, please fix the tests while you are at it.

Rajeshpatel07 commented:
Okay thanks, and about the cassette folder yaml files. When running tkts test file it is giving error `can't overwrite cassette files`. 

And the other unnecessary changes are added by the my code editor auto format.

gridhead commented:
The https://github.com/fedora-infra/pagure-exporter/pull/172/commits/fc1c9aff30a52efe5555063b0169808605e39556 commit should not have been introduced if the unnecessary changes had not been made. Please ensure that the unnecessary commits are squashed into one. 

Regarding the environment variables, please create those from your Pagure and GitLab accounts depending on which repositories you are trying to migrate over using the tool.

Rajeshpatel07 commented:
@gridhead I have squashed commit and added `env` keys in github workflow. But I have some doubts can you please clarify

1. In some test cases the log are hard code with your name like this
```python
pytest.param(
            f"--pfm {envr['TEST_PFM']} --srce {envr['TEST_SRCE']} --dest {envr['TEST_DEST']} --pkey {envr['TEST_PKEY']} --gkey {envr['TEST_GKEY']} --fusr {envr['TEST_FUSR']} --tusr {envr['TEST_TUSR']} tkts --status SHUT --series",  # noqa: E501
            0,
            [
                "[ WARN ] Extracting all closed issue tickets without labels, without states, without privacy, without comments and with order off the given selection",  # noqa: E501
                "Found 2 issue ticket(s) across 1 page(s) in ",
                "[ BUSY ] Migrating issue ticket without labels #2 'This is the title of the second test issue' by 'Akashdeep Dhar (ID t0xic0der)'...",  # noqa: E501
                "Migrated to ",
                "[ BUSY ] Migrating issue ticket without labels #4 'This is the title of the fourth test issue' by 'Akashdeep Dhar (ID t0xic0der)'...",  # noqa: E501
                "Migrated to ",
            ],
            id="Transferring issue tickets with SHUT status without labels, without states, without privacy, without comments and with order",  # noqa: E501
        ),
```

 If I run the test with my `api` keys the test will fail for sure. Is this written like this for a specific purpose. So can you please clarify.


2. The tool `python-recorder` is creating `cassettes/*.yml` files which hold the api responses. When I run tests with my `api` keys every test related to `tkts` is giving error and failing.

This is the error message
```bash
ERROR test/test_tkts.py::test_main_tkts[Transferring issue tickets with FULL status without labels, without states, without privacy, without comments and with order the identities of which fall in the given selection] - vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ('/home/raj...
```

Should I overwrite the files or not.

Rajeshpatel07 commented:
@gridhead  I am closing this PR as the issue is closed.
